envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -126,34 +127,43 @@ fun Value.toIncomingEndpoint(): IncomingEndpoint {
     }
 }
 
-private fun Value?.toIncomingTimeoutPolicy(): Incoming.TimeoutPolicy {
-    val idleTimeout: Duration? = this?.field("idleTimeout")?.stringValue
-        ?.takeIf { it.isNotBlank() }
-        ?.let { Durations.parse(it) }
-    val responseTimeout: Duration? = this?.field("responseTimeout")?.stringValue
-        ?.takeIf { it.isNotBlank() }
-        ?.let { Durations.parse(it) }
-
-    val connectionIdleTimeout: Duration? = this?.field("connectionIdleTimeout")?.stringValue
-        ?.takeIf { it.isNotBlank() }
-        ?.let { Durations.parse(it) }
+fun Value?.toIncomingTimeoutPolicy(): Incoming.TimeoutPolicy {
+    val idleTimeout: Duration? = parseToDuration(this?.field("idleTimeout"))
+    val responseTimeout: Duration? = parseToDuration(this?.field("responseTimeout"))
+    val connectionIdleTimeout: Duration? = parseToDuration(this?.field("connectionIdleTimeout"))
 
     return Incoming.TimeoutPolicy(idleTimeout, responseTimeout, connectionIdleTimeout)
 }
 
 private fun Value?.toOutgoingTimeoutPolicy(properties: SnapshotProperties): Outgoing.TimeoutPolicy {
-    val idleTimeout: Duration? = this?.field("idleTimeout")?.stringValue
-        ?.takeIf { it.isNotBlank() }
-        ?.let { Durations.parse(it) }
+    val idleTimeout: Duration? = parseToDuration(this?.field("idleTimeout"))
         ?: Durations.fromMillis(properties.egress.commonHttp.idleTimeout.toMillis())
-    val requestTimeout: Duration? = this?.field("requestTimeout")?.stringValue
-        ?.takeIf { it.isNotBlank() }
-        ?.let { Durations.parse(it) }
+    val requestTimeout: Duration? = parseToDuration(this?.field("requestTimeout"))
         ?: Durations.fromMillis(properties.egress.commonHttp.requestTimeout.toMillis())
 
     return Outgoing.TimeoutPolicy(idleTimeout, requestTimeout)
 }
 
+private fun parseToDuration(timeout: Value?): Duration? {
```

Comment:
This looks like a utility function that does not need to be coupled with `NodeMetadata`, therefore it should land in it's own space as a public function that can be unit tested.